src/main/java/com/fasterxml/jackson/databind/util/EnumResolver.java

```diff
@@ -81,6 +81,50 @@ public static EnumResolver constructFor(DeserializationConfig config,
                 config.isEnabled(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS));
     }
 
+    /**
+     * @since 2.15
+     */
+    public static EnumResolver constructFor(DeserializationConfig config, Class<?> enumCls0, AnnotatedClass annotatedClass) {
+        return _constructFor(config, enumCls0, annotatedClass, config.getAnnotationIntrospector(),
+            config.isEnabled(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS));
+    }
+
+    /**
+     * @since 2.15
+     */
+    public static EnumResolver _constructFor(DeserializationConfig config, Class<?> enumCls0,
```

Comment:
Aside from combining `Class` & `AnnotatedClass`, can just pass `config` and drop `AnnotationIntrospector` argument (can use `config.getAnnotationIntrospector()`).

Reply:
Yes, although second one is trickier as it is possible there will be more complicated logic to determine case-sensitivity (not just global, but per-property overrides). So caller may need to figure it out. But AnnotationIntrospector definitely need not be passed separately.